### PR TITLE
Adding geolocation string formats to email magic link OTP terminal view

### PR DESCRIPTION
**Pull request by justinabrokwah-okta**
## Description:
- In [previous PR](https://github.com/okta/okta-signin-widget/pull/2235), geolocation was only shown if the response contained:
  - `city`
  - `state` OR `country` 
- ~~Now, geolocation is displayed if `city`, `state`, or `country` are provided. If multiple are provided, then they are all shown.~~
- Thanks to i18n team and discussion in comments below, only display location in the following formats:
  - `City, State, Country`
  - `City, Country`

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
- Collection of screenshots here: https://okta.box.com/s/ahc83rg3eq6y1gziph4pqiknh0z33osq
  - Showing different formats

### Issue:

- [OKTA-444841](https://oktainc.atlassian.net/browse/OKTA-444841)

### CODE REVIEW

File: packages/@okta/i18n/src/properties/login.properties
Comment (jmelberg-okta) [inline packages/@okta/i18n/src/properties/login.properties:1276]: Not an expert with denoting order, but we may want to further segment these into categories **if order changes** based on the input type.

@evelynteo-okta - Are you aware of any ordering changes between various languages for `City, State`, `State, Country`, or `City, Country`? I would assume the more focused area will always be the first positional argument, but am not 100% sure if this is correct.

File: src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js
Comment (jmelberg-okta) [inline src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js:4]: style-nit: Consider adding a default function param to avoid the large `if (location)` block below.

File: src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js
Comment (jmelberg-okta) [inline src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js:28]: I doubt customers will ever want to override these keys, but can we generalize these to avoid the long names?

I'd suggest:

```suggestion
      geolocation = loc('geolocation.formatting.partial', 'login', values);
      break;
    case 3:
      geolocation = loc('geolocation.formatting.full', 'login', values);
```

**Review by jmelberg-okta — APPROVED** [review]
Great work!

Left a couple of questions and suggestions I think we should try to answer before closing this PR.

File: test/testcafe/spec/TerminalView_spec.js
Comment (pradeepdewda-okta) [inline test/testcafe/spec/TerminalView_spec.js:157]: Nice work adding these tests.. But i would suggest to add these test to specific view. for eg in this it should be `EmailMagicLinkOTPTerminalView.js`.. 

Keeping TerminalView spec clean can help other devs who are not aware of email magic link related tests.

File: packages/@okta/i18n/src/properties/login.properties
Comment (pradeepdewda-okta) [reply packages/@okta/i18n/src/properties/login.properties:1276]: Do we even need i18n for country names.. I think we are passing whatever coming from server.. so server should have already localized it right ?

Comment (jmelberg-okta) [reply packages/@okta/i18n/src/properties/login.properties:1276]: @pradeepdewda-okta - We need to account for ordering of the geolocation, which needs to be handled by clients. [Not all languages](https://en.wikipedia.org/wiki/Chinese_punctuation) use commas for lists/separating values, so placing punctuation within our i18n resource files is required.

Comment (jmelberg-okta) [reply packages/@okta/i18n/src/properties/login.properties:1276]: @nikhilvenkatraman-okta - FYI ☝🏼

File: test/testcafe/spec/TerminalView_spec.js
Comment (justinabrokwah-okta) [reply test/testcafe/spec/TerminalView_spec.js]: Totally agree, just changed it

File: src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js
Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js]: Updated!

File: packages/@okta/i18n/src/properties/login.properties
Comment (justinabrokwah-okta) [reply packages/@okta/i18n/src/properties/login.properties]: @jmelberg-okta Great catch, looking forward to what the i18n team thinks

Comment (evelynteo-okta) [reply packages/@okta/i18n/src/properties/login.properties]: @jmelberg-okta @nikhilvenkatraman-okta @justinabrokwah-okta @pradeepdewda-okta  `City, State, State, Country`, or `City, Country` --- **Yes, the ordering , groupings, styling (uppercase or lowercase) and punctuation of each piece of information, will change depending on the locale and country.** 

For example:
Address-Format

Postal codes, in countries that have them, are usually numeric, sometimes containing a space or a hyphen. Different countries use different terms for postal codes (e.g. Zone Improvement Plan, Postleitzahl) and different abbreviations for the terms (e.g. ZIP, PLZ, CEP, CAP, PIN).


Reference: 

International Component for Unicode (ICU) Address Format implementation at Salesforce: https://help.salesforce.com/s/articleView?id=sf.admin_supported_locales.htm&type=5

Comment (justinabrokwah-okta) [reply packages/@okta/i18n/src/properties/login.properties]: Updated code to display location in these formats: `City, State, Country` or `City, Country`, preferring the former if the state is in the response

**Review by pradeepdewda-okta — APPROVED** [review]

File: src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js
Comment (hanyuanzhang-okta) [inline src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js:12]: Is PM aware that we will display city, state and country in this case?

Comment (justinabrokwah-okta) [reply src/v2/view-builder/views/consent/EmailMagicLinkOTPTerminalView.js:12]: Yup, I've dropped a note to Bhagya, waiting for her response before final merge

**Review by hanyuanzhang-okta — APPROVED** [review]
LGTM!

**Pull request closed without merging**
